### DISCUSSION TITLE
Can I push data to Budibase though an API?

### DISCUSSION THREAD

**Discussion by tudisco** [Budibase/budibase#3425] [Help]
Could I, for example, use an API provided by Budibase to add data from a contact form on a website to Budibase?

**Comment by mjashanks** [comment]
This is technically possible. However, our API is not well documented at the moment.

An easier solution would be to build the form with budibase and embed it in an iFrame
1. Create a new screen with access level = `Public` (this will be the form on your website)
2. Set it's layout to "Empty Layout", so there is no navigation
3. Build up your form manually
4. On the `Save` / `Submit` button, add a "Trigger Automation" action
5. Add a field for every field on your form - and bind the values to your form

![image](https://user-images.githubusercontent.com/3524181/142390626-07d8066f-eec9-4e35-a6ac-c0b2700cb29a.png)

6. Go to the `Automations` section and find your new automation
7. Add a `Save Row` step that saves the values sent to the automation to a row in a table.

Now, on your website, add an iFrame and point it to the URL of your new screen. You will need to publish your app first.
